diagnostic_aggregator/example/example_pub.py

```diff
@@ -53,10 +53,15 @@ class DiagnosticTalker(Node):
 
     def __init__(self):
         super().__init__('diagnostic_talker')
+        self.declare_parameter('publish_transient_local', False)
+        use_transient_local = self.get_parameter('publish_transient_local').get_parameter_value().bool_value
+        qos = qos_profile_system_default
+        if use_transient_local:
+            qos.durability = QoSDurabilityPolicy.TRANSIENT_LOCAL
         self.i = 0
         self.pub = self.create_publisher(DiagnosticArray,
                                          '/diagnostics',
-                                         qos_profile_system_default)
```

Comment:
I think you need two topics to subscribe -- one supporting legacy (backwards compatible) on `/diagnostics` and another supporting stateful on `/diagnostics_ros2`. This is especially important until the diagnostic updater is also updated to support these QoS policies.

Reply:
yeah, the way it currently is (backwards compatibility handled by solely a parameter) is not going to work for the use case. so I will do the two topic way.